Core/include/Acts/Propagator/ConstrainedStep.hpp

```diff
@@ -45,7 +45,7 @@ class ConstrainedStep {
   /// from navigator - this would be a navigation step
   /// from actor     - this would be an actor condition
   /// from user      - this is user given for what reason ever
-  enum Type : int { navigator = 0, actor = 1, user = 2 };
+  enum class Type : int { Navigator = 0, Actor = 1, User = 2 };
```

Comment:
In the end you decided against the `using enum`?

Reply:
ooops - on its way

Reply:
`enum class` conflicts with the current implementation because we rely on implicit integer conversion

Using some helpers might improve the situation but we can also just continue using `enum`. What do you think @paulgessinger ?

Reply:
I personally think explicit conversion of `enum class` with `toUnderlying` is preferable if there are not too many places in the code that require this. 

However, I do concede that this has downsides over sticking to plain `enum`.

Iirc we only use it to index into the step constraint array inside the `ConstrainedStep` object. If that's the case, I see the explicit static cast as acceptable.

Reply:
I applied the changes again using `toUnderlying `. I don't think we really lose any readability using the helpers.